File: Frontend/src/Models/MessageModel.ts
/** Allowed message authors in the chat UI and API contract. */
export type SenderType = "user" | "assistant";

/** TypeScript model for a single chat message. */
export interface MessageModel {
  id: string;
  conversationId: string;
  senderType: SenderType;
  content: string;
  createdAt: string;
}

File: Backend/src/models/message_model.py
"""Pydantic models representing chat messages."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class MessageModel(BaseModel):
    """Message object returned to API consumers."""

    id: str
    conversationId: str
    senderType: Literal["user", "assistant"]
    content: str
    createdAt: datetime


class CreateMessageModel(BaseModel):
    """Internal model for creating a message record."""

    conversationId: str
    senderType: Literal["user", "assistant"]
    content: str
